sycl/include/CL/sycl/detail/device_filter.hpp

```diff
@@ -0,0 +1,96 @@
+//==---------- device_filter.hpp - SYCL device filter descriptor -----------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/backend_types.hpp>
+#include <CL/sycl/detail/defines.hpp>
+#include <CL/sycl/info/info_desc.hpp>
+
+#include <iostream>
+#include <string>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+
+struct device_filter {
+  backend Backend = backend::host;
```

Comment:
do we really what HOST to be our default backend. Isn't it better to use all here to select any available backend?

Reply:
Thanks. Done.